### ISSUE TITLE
[Bug] String interpolation does not work when overwriting a list as another list from a different config group

### ISSUE THREAD

**Issue by apugoneappu** [facebookresearch/hydra#1539]
# 🐛 Bug
## Description
For a key with a list as its value defined in a config_group, I get an error when trying to overwrite that key from another config_group as a list using string interpolation. This error does not happen when the original key is overwritten to a regular string using string interpolation. 

## Checklist
- [x] I checked on the latest version of Hydra
- [x] I created a minimal repro (See [this](https://stackoverflow.com/help/minimal-reproducible-example) for tips).

## To reproduce

**Minimal Code/Config snippet to reproduce**

configs/config.yaml
```
foo: 55

defaults:
- first_group: a
- second_group: b
```

configs/first_group/a.yaml
```
# @package _group_

a_list:
- first_item
- second_item
```

configs/second_group/b.yaml
```
# @package _global_

# Does not cause an error
# first_group:
#   a_list: ${foo}

# Causes an error
first_group:
  a_list:
    - ${foo}
```

**Stack trace/error message**

Short error message
```
Error merging second_group=b
str interpolation key 'foo' not found
        full_key: first_group.a_list[0]
        reference_type=Optional[List[Any]]
        object_type=list
```

<details>
<summary>
stack trace (click to unfold)
</summary>

```
Traceback (most recent call last):
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 720, in _merge_config
    ret = OmegaConf.merge(cfg, loaded_cfg)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/omegaconf.py", line 321, in merge
    target.merge_with(*others[1:])
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 331, in merge_with
    self._format_and_raise(key=None, value=None, cause=e)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/base.py", line 101, in _format_and_raise
    type_override=type_override,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/_utils.py", line 629, in format_and_raise
    _raise(ex, cause)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/_utils.py", line 610, in _raise
    raise ex  # set end OC_CAUSE=1 for full backtrace
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 329, in merge_with
    self._merge_with(*others)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 347, in _merge_with
    BaseContainer._map_merge(self, other)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 294, in _map_merge
    dest_node._merge_with(src_value)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 347, in _merge_with
    BaseContainer._map_merge(self, other)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 294, in _map_merge
    dest_node._merge_with(src_value)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 367, in _merge_with
    for item in other:
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/listconfig.py", line 457, in __next__
    v = self.lst[self.index]
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/listconfig.py", line 176, in __getitem__
    self._format_and_raise(key=index, value=None, cause=e)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/base.py", line 101, in _format_and_raise
    type_override=type_override,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/_utils.py", line 694, in format_and_raise
    _raise(ex, cause)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/omegaconf/_utils.py", line 610, in _raise
    raise ex  # set end OC_CAUSE=1 for full backtrace
omegaconf.errors.ConfigKeyError: str interpolation key 'foo' not found
        full_key: first_group.a_list[0]
        reference_type=Optional[List[Any]]
        object_type=list

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "main.py", line 11, in <module>
    run_app()
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/main.py", line 37, in decorated_main
    strict=strict,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/utils.py", line 347, in _run_hydra
    lambda: hydra.run(
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/utils.py", line 201, in run_and_report
    raise ex
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/utils.py", line 198, in run_and_report
    return func()
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/utils.py", line 350, in <lambda>
    overrides=args.overrides,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/hydra.py", line 104, in run
    run_mode=RunMode.RUN,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/hydra.py", line 512, in compose_config
    from_shell=from_shell,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 156, in load_configuration
    from_shell=from_shell,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 262, in _load_configuration
    run_mode=run_mode,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 805, in _merge_defaults_into_config
    hydra_cfg = merge_defaults_list_into_config(hydra_cfg, user_list)
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 783, in merge_defaults_list_into_config
    package_override=default1.package,
  File "/Users/apoorve/anaconda3/envs/adven/lib/python3.7/site-packages/hydra/_internal/config_loader_impl.py", line 726, in _merge_config
    ) from ex
hydra.errors.ConfigCompositionException: Error merging second_group=b
```
</details>

## Expected Behavior
I expected the final config to be like:

```json
foo: 55
first_group:
  a_list: [55]
```

## System information
- **Hydra Version** :  1.0.6
- **Python version** : 3.7.9
- **Virtual environment type and version** : Anaconda 4.9.2
- **Operating system** : macOS Big Sur 11.2.3 (20D91)

## Additional context
The problem does not occur if the key is not a list in either a.yaml or b.yaml

**Comment by apugoneappu** [comment]
Please see this minimal code to reproduce the error.

[test_hydra.zip](https://github.com/facebookresearch/hydra/files/6264474/test_hydra.zip)

**Comment by omry** [comment]
Thanks for reporting. Looks like this is already fixed in Hydra 1.1.
Please try to upgrade to the dev release.

**Comment by apugoneappu** [comment]
Thanks for all the great work @omry! Is there any place I could see the bugs that have been fixed in Hydra 1.1?
I have already looked at https://hydra.cc/docs/next/upgrades/1.0_to_1.1/* but this does not list the small bug fixes I guess.

**Comment by jieru-hu** [comment]
> Thanks for all the great work @omry! Is there any place I could see the bugs that have been fixed in Hydra 1.1?
> I have already looked at https://hydra.cc/docs/next/upgrades/1.0_to_1.1/* but these pages do not list the small bug fixes I guess.

Hi! here is the [list of issues](https://github.com/facebookresearch/hydra/issues?q=is%3Aissue+milestone%3A%22Hydra+1.1.0%22+is%3Aclosed) closed with 1.1

**Comment by omry** [comment]
The upgrades section is more guidance about migration.
Since both Hydra 1,1 and OmegaConf 2.1 are not yet released, there are no release notes.
RC versions are expected within a few weeks.

You can generate release notes locally with towncrier though.
e.g:
```
~/dev/hydra$ towncrier  --draft | pastebinit 
```

Hydra 1.1: http://paste.ubuntu.com/p/7GpkDVgRPB/
OmegaConf 2.1: http://paste.ubuntu.com/p/mJ5S64F8H2/

**Comment by apugoneappu** [comment]
Thanks for the help!

**Comment by omry** [comment]
The specific bug you ran into is mentioned in the OmegaConf generated release notes:
```
Fix bug where interpolations were unnecessarily resolved during merge ([#431](https://github.com/omry/omegaconf/issues/431))
```
